.github/workflows/build.yaml

```
@@ -0,0 +1,191 @@
+
+          echo "📦 Artifact: $ARTIFACT"
+          echo "📤 S3 URI: $S3_URI"
+
+          if aws s3api head-object --bucket "$S3_BUCKET" --key "$S3_KEY" 2>/dev/null; then
+            echo "BUILD_RESULT=skipped" >> $GITHUB_ENV
+            echo "✔️ Artifact already exists in S3. Skipping build."
+          else
+            echo "BUILD_RESULT=built" >> $GITHUB_ENV
+            echo "🚧 Building artifact for $MATRIX_OS..."
+            mvn clean package -P${MATRIX_OS} -Dbuild.revision="$BUILD_VERSION"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)